src/priors.jl

```diff
@@ -0,0 +1,41 @@
+abstract type Prior end
+
+struct DefaultPrior <: Prior end
+
+function DefaultPrior(x...)
+    throw(ArgumentError("DefaultPrior accepts no parameters, please use CustomPrior"))
+    return nothing
+end
+
```

Comment:
```suggestion
```
Isn't this error already provided pretty well by Julia? Or, at least this constructor should also take keyword arguments to cover all (or are there more?) bases.
```
julia> struct A end

julia> A()
A()

julia> julia> A(122)
ERROR: MethodError: no method matching A(::Int64)
Closest candidates are:
  A() at REPL[1]:1
Stacktrace:
 [1] top-level scope
   @ REPL[3]:1